src/ProgressOnderwijsUtils/SingleSignOn/SsoProcessor.cs

```diff
@@ -88,28 +88,34 @@ static string EncodeQueryParameter(string key, string value)
                 return Maybe.Error("Missing attribute " + UID);
             }
 
+            var (inresponseTo, notOnOrAfter) = GetSubjectConfirmationData(assertion);
+
+            var now = DateTime.UtcNow;
+            if (now >= notOnOrAfter) {
+                return Maybe.Error($"Expired: {now} >= {notOnOrAfter}");
+            }
+
             return Maybe.Ok(
                 new SsoAttributes {
                     uid = uid,
                     domain = GetNullableAttribute(assertion, DOMAIN),
                     email = GetAttributes(assertion, MAIL),
                     roles = GetAttributes(assertion, ROLE),
-                    InResponseTo = GetInResponseTo(assertion),
-                    IssueInstant = (DateTime)assertion.Attribute("IssueInstant"),
+                    InResponseTo = inresponseTo,
                     AuthnContextClassRef = (string)authnStatement.Element(SamlNamespaces.SAML_NS + "AuthnContext").Element(SamlNamespaces.SAML_NS + "AuthnContextClassRef"),
                 });
         }
 
-        static string? GetInResponseTo(XElement assertion)
+        static (string? inresponseTo, DateTime notOnOrAfter) GetSubjectConfirmationData(XElement assertion)

```

Comment:
```suggestion
        static (string? inresponseTo, DateTime validBefore) GetSubjectConfirmationData(XElement assertion)
```

of 

```suggestion
        static (string? inresponseTo, DateTime validUntil) GetSubjectConfirmationData(XElement assertion)
```

leest mogelijk makkelijker?

Reply:
Die naam heb ik niet bedacht; zo heet het gewoon.

Reply:
fair enough!